modules/local/prefetchchecker/main.nf

```diff
@@ -0,0 +1,23 @@
+process PREFETCH_CHECKER {
+    tag "prefetch_checker"
+    label 'process_low'
+
+    input:
+    val failures  // list of failures
+
+    output:
+    path("failures_report.csv"), emit: failure_report
+
+    exec:
+    task.workDir.resolve("failures_report.csv").withWriter { writer ->
+        // Failures
+        if (failures.size() > 0) {
+            failures.each { writer.writeLine "${it[0].id},${it[1]}" }
```

Comment:
Could you write a header line that gives the column names? This should also let you avoid the else statement writing an empty line to the CSV (since you will have already written a header line).

Reply:
Updated in https://github.com/phac-nml/fetchdatairidanext/commit/577e50d4d5115d653306e0836acab28bdcf25a1f